merino/web/api_v1.py

```diff
@@ -114,6 +117,28 @@ async def suggest(
     return JSONResponse(content=jsonable_encoder(response))
 
 
+def emit_suggestions_per_metrics(
+    metrics_client: Client,
+    suggestions: list[BaseSuggestion],
+    searched_providers: list[BaseProvider],
+) -> None:
+    """Emit metrics for suggestions per request and suggestions per request by provider."""
+    metrics_client.histogram("suggestions-per.request", value=len(suggestions))
+
+    suggestion_counter = Counter(suggestion.provider for suggestion in suggestions)
+
+    logging.error(f"{searched_providers}")
+
+    for provider in searched_providers:
+        provider_name = provider.name
+        suggestion_count = suggestion_counter[provider_name]
+        metrics_client.histogram(
+            "suggestions-per.provider",
```

Comment:
By convention, we normally write the provider id as part of the metric name.

 ```suggestion
            f"suggestions-per.provider.{provider_name}",
```

Reply:
Ah, if we're providing this in the metric name, then we should remove it from the tags? I agree that this is more conventional, and would prefer this over using tags for the provider. The only downside is that cross comparisons will be more difficult, but I'm not sure this is useful to look at.

Reply:
I also prefer this. for context, i wrote the ticket based on the way this metric was implemented in the rust version of merino.

Reply:
I think either format is fine as long as it's consistent across the code base. If we prefer tags, could you file a follow-up ticket to convert the existing metrics as well (and update the Grafana dashboard)? Thank!

Reply:
By popular opinion, I've renamed the metric to use the suggested naming scheme. I'll update the Grafana dashboard once this change is out!